engine/src/GraphicsController.cpp

```diff
@@ -90,7 +89,83 @@ void GraphicsController::draw_skybox(const resources::Shader *shader, const reso
     CHECKED_GL_CALL(glBindTexture, GL_TEXTURE_CUBE_MAP, skybox->texture());
     CHECKED_GL_CALL(glDrawArrays, GL_TRIANGLES, 0, 36);
     CHECKED_GL_CALL(glBindVertexArray, 0);
-    CHECKED_GL_CALL(glDepthFunc, GL_LESS); // set depth function back to default
+    CHECKED_GL_CALL(glDepthFunc, GL_LESS);
     CHECKED_GL_CALL(glBindTexture, GL_TEXTURE_CUBE_MAP, 0);
 }
+
+void GraphicsController::draw_instancing(const resources::Shader *shader, const resources::Instancing *instancing) {
+    glm::mat4 view = m_camera.view_matrix();
+    shader->use();
+    shader->set_mat4("view", view);
+    shader->set_mat4("projection", projection_matrix<>());
+    shader->set_int("texture_diffuse1", 0);
```

Comment:
[CRITICAL] Instanciranje neće raditi ukoliko se tekstura u šejderu zove drugačije. Modul `engine` bi trebalo da bude što nezavisniji od bilo kakve konkretne aplikacije ili njene implementacije, uključujući i kako će korisnik modula `engine` odabrati da nazove teksture. 

Možete pogledati skybox kao primer. Skybox nigde ne pretpostavlja da će se cubemap tekstura zvati nekako određeno već samo učita potrebne resurse a na aplikaciji je onda da implementira crtanje.